corehq/motech/value_source.py

```diff
@@ -73,15 +73,15 @@ def check_direction(self, direction):
         imported or exported.
 
         >>> value_source = ValueSource(direction=DIRECTION_BOTH)
-        >>> bool(value_source.check_direction(DIRECTION_EXPORT))
+        >>> value_source.check_direction(DIRECTION_EXPORT)
         True
 
         >>> value_source = ValueSource(direction=DIRECTION_IMPORT)
-        >>> bool(value_source.check_direction(DIRECTION_EXPORT))
+        >>> value_source.check_direction(DIRECTION_EXPORT)
         False
 
         """
-        return DIRECTIONS[self.direction] & DIRECTIONS[direction]
+        return not self.direction or direction == self.direction
```

Comment:
this should be `and`, not `or`. Also, It would give an error when comparing with `None` i believe.

Reply:
I'm not sure I understand.

`self.direction` is `None` when the integration should both import and export values, in which case this method should return `True` regardless of the value of `direction`. If `self.direction` is "out" then it must return `True` if `direction` is also "out", and likewise for "in". i.e:
```python
>>> other_direction = "in"
>>> self_direction = None
>>> not self_direction or other_direction == self_direction
True
>>> self_direction = "out"
>>> not self_direction or other_direction == self_direction
False
>>> self_direction = "in"
>>> not self_direction or other_direction == self_direction
True
```

Reply:
oh yes. You are right.